coretex/entities/sample/image_sample/image_sample.py

```diff
@@ -69,6 +75,42 @@ def saveAnnotation(self, coretexAnnotation: CoretexImageAnnotation) -> bool:
         response = networkManager.post("session/save-annotations", parameters)
         return not response.hasFailed()
 
+    def saveMetadata(self, metadata: Dict[str, Any]) -> None:
+        """
+            Saves a json object as metadata for the sample
+
+            Parameters
+            ----------
+            metadata : dict[str, Any]
+                Json object containing sample metadata
+
+            Raises
+            ------
+            NetworkRequestError -> if metadata upload failed
+        """
+
+        parameters = {
+            "id": self.id,
+            "data": metadata
+        }
+
+        response = networkManager.post("session/save-metadata", parameters)
+        if response.hasFailed():
+            raise NetworkRequestError(response, f"Failed to upload metadata for sample {self.name}")
+
+    def loadMetadata(self) -> Dict[str, Any]:
+        """
+            Loads sample metadata into a dictionary
+        """
+
+        try:
+            with self.metadataPath.open("r") as metadataFile:
+                metadata: Dict[str, Any] = json.load(metadataFile)
+                return metadata
+        except FileNotFoundError:
+            raise FileNotFoundError(f"Metadata file \"{self.metadataPath}\" was not found")
```

Comment:
This seems hacky - catching an exception then raising the same one but with a different message. Why not check if `metadataPath` exists before trying to load it? This way there is no need for you to use try-catch block.